app/controllers/OrganizationController.scala

```diff
@@ -71,4 +78,36 @@ class OrganizationController @Inject()(organizationDAO: OrganizationDAO,
   def getOperatorData: Action[AnyContent] = Action {
     Ok(Json.toJson(conf.operatorData))
   }
+
+  def update(organizationName: String): Action[JsValue] = sil.SecuredAction.async(parse.json) { implicit request =>
+    withJsonBodyUsing(organizationUpdateReads) {
+      case (displayName, newUserMailingList) =>
+        for {
+          organization <- organizationDAO.findOneByName(organizationName) ?~> Messages("organization.notFound",
+                                                                                       organizationName) ~> NOT_FOUND
+          _ <- bool2Fox(request.identity.isAdminOf(organization._id)) ?~> "notAllowed" ~> FORBIDDEN
+          _ <- organizationDAO.updateFields(organization._id, displayName, newUserMailingList)
+          updated <- organizationDAO.findOne(organization._id)
+          organizationJson <- organizationService.publicWrites(updated)
+        } yield Ok(organizationJson)
+    }
+  }
+
+  def delete(organizationName: String): Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    for {
+      organization <- organizationDAO.findOneByName(organizationName) ?~> Messages("organization.notFound",
+                                                                                   organizationName) ~> NOT_FOUND
+      _ <- bool2Fox(request.identity.isAdminOf(organization._id)) ?~> "notAllowed" ~> FORBIDDEN
```

Comment:
Does this differ from `assertUpdateAccess` which is used in the `Organization` below? Can this be unified?

Reply:
You are right, this is a level of redundancy, and this is kind of a pattern across the wK back-end.
The controllers run access checks first, before any side effects can occur. So the update access checks of the DAOs are kind of unnecessary.
I created the follow-up issue https://github.com/scalableminds/webknossos/issues/5327 for this refactoring, since it does not only concern the organizationDAO.